### ISSUE TITLE
[Testing] memory_format decorator

### ISSUE THREAD

**Issue by cpuhrsch** [pytorch/pytorch#62925]
## 🚀 Feature
We currently have decorators that allow us to parameterize tests on dtypes, devices, etc., but I don't see one for memory formats (i.e. torch.channels_last and torch.contiguous_format).

## Motivation

There are multiple operators where we want to check support for and propagation of these memory formats. There were some recent issues where [BatchNorm2D](https://github.com/pytorch/pytorch/pull/62773) and fused [conv2d_relu, conv2d_add_relu](https://github.com/pytorch/pytorch/pull/62482) aren't propagating them or are setting them incorrectly for the result Tensor which can cause performance regressions or errors. Having this decorator allows us to write separate tests for them and increase coverage.

## Pitch

Add @memoryFormat decorator to our suite of test decorators

## Alternatives

The usual copy-pasting for for-loops within tests.

## Additional context

N/A

**Comment by mruberry** [comment]
cc @jbschlosser who's working on a test parametrization prototype
